builtin/providers/google/resource_sql_database.go

```diff
@@ -121,6 +124,7 @@ func resourceSqlDatabaseDelete(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	err = sqladminOperationWait(config, op, "Delete Database")
+	googleMutexKV.Unlock(instanceMutexKey(project, instance_name))
```

Comment:
Same, would really like to see the `defer` approach used here, for completeness.